public/Get-DbaReplServer.ps1

```diff
@@ -60,9 +60,14 @@ function Get-DbaRepServer {
         if (Test-FunctionInterrupt) { return }
         foreach ($instance in $SqlInstance) {
             try {
-                # use System.Data instead of Microsoft.Data
-                $sqlconn = New-SqlConnection -SqlInstance $instance -SqlCredential $SqlCredential
-                New-Object Microsoft.SqlServer.Replication.ReplicationServer $sqlconn
+                $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+                $replServer = New-Object Microsoft.SqlServer.Replication.ReplicationServer
+                $replServer.ConnectionContext = $Server.ConnectionContext.SqlConnectionObject
+                $replServer | Add-Member -Type NoteProperty -Name ComputerName -Value $server.ComputerName -Force
+                $replServer | Add-Member -Type NoteProperty -Name InstanceName -Value $server.ServiceName -Force
+                $replServer | Add-Member -Type NoteProperty -Name SqlInstance -Value $server.DomainInstanceName -Force
+
+                Select-DefaultView -InputObject $replServer -Property ComputerName, InstanceName, SqlInstance, DistributorInstalled, DistributorAvailable, IsDistributor, IsPublisher, HasRemotePublisher, DistributionServer, DistributionDatabase, WorkingDirectory, AgentCheckupInterval, DistributionDatabases, DistributionPublishers, ReplicationDatabases, RegisteredSubscribers
```

Comment:
Not sure if there is possibility to lessen number of properties? Once you get over 7 (think) PowerShell converts it to a list. Consider what is bare minimum they need to know if they are running this interactively? I'd say: ComputerName, InstanceName, SqlInstance, IsDistributor, IsPublisher, DistributionServer, DistributionDatabase is sufficient for a "server" object return.

Reply:
Makes sense to me..
(Also the ReplicationDatabases include all the databases on the instance so is a bit confusing anyway tbh)